docs/source/debugging.md

```diff
@@ -0,0 +1,183 @@
+# Debugging and Profiling
+
+While using the hi-ml toolbox modules, you might encounter some errors that require running the code step by step in
+order to track down the source of these errors. Here are some guidelines to help you debug and/or profile hi-ml deep
+learning training pipelines.
+
+## Debugging within VS Code
+
+VS code has a great [Debugging Support](https://code.visualstudio.com/docs/editor/debugging) for many programming
+languages. Make sure to install and enable the [Python Extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+to debug hi-ml toolbox modules built in Python.
+
+### Debugging configs
+
+The hi-ml repository is organised in
+[Multi-root workspaces](https://code.visualstudio.com/docs/editor/workspaces#_multiroot-workspaces) to account
+for environment differences among the modules and offer flexibility to configure each module seperatly.
+
+We provide a set of example debugging configs for each of hi-ml module:
+
+* [launch.json in hi-ml](https://github.com/microsoft/hi-ml/tree/main/hi-ml/.vscode/lanch.json)
+* [launch.json in hi-ml-azure](https://github.com/microsoft/hi-ml/tree/main/hi-ml-azure/.vscode/lanch.json)
+* [launch.json in hi-ml-histopathology](https://github.com/microsoft/hi-ml/tree/main/hi-ml-histopathology/.vscode/lanch.json)
+
+VS Code restricts debugging to user-written code only by default. If you want to step through external code and
+standard libraries functions, set `"justMyCode": false` inside the debugging config block in the `launch.json` file.
+
+In particular, if you would like to debug the current file while breaking through `pytorch` code, navigate to
+`himl-projects.code-workspace` in the repo root and edit the "launch" block as follow:
+
+```json
+"launch": {
+    "configurations": [
+      {
+        "name": "Python: Current File",
+        "type": "python",
+        "request": "launch",
+        "program": "${file}",
+        "console": "integratedTerminal",
+        "justMyCode": false
+      },
+    ],
+}
+```
+
+### Pytorch Lightning flags for debugging and quick runs
```

Comment:
Nit: PyTorch with capital T (there are a few other references to it on this page too)